caddyhttp/push/handler_test.go

```diff
@@ -307,6 +308,58 @@ func TestMiddlewareShouldInterceptLinkHeaderPusherError(t *testing.T) {
 	comparePushedResources(t, expectedPushedResources, pushingWriter.pushed)
 }
 
+func TestMiddlewareShouldPushIndexFile(t *testing.T) {
+	// given
+	indexFile := "/index.html"
+	request, err := http.NewRequest(http.MethodGet, "/", nil) // Request root directory, not indexfile itself
+	if err != nil {
+		t.Fatalf("Could not create HTTP request: %v", err)
+	}
+
+	root := "."
```

Comment:
Let's use a temporary folder instead. See ioutil.TempDir for an easy way to do this.